src/routes/channel.rs

```diff
@@ -26,6 +26,32 @@ use {
     tokio_stream::wrappers::BroadcastStream,
 };
 
+// Centralized deserialization function to handle both old and new formats
+pub fn deserialize_channel_content(content: &str) -> Result<ChannelContent> {
```

Comment:
Instead of needing this method, you can use Serde's [untagged enums](https://serde.rs/enum-representations.html#untagged) which let you model multiple possible types and then serde will try and parse it against both types, failing only if it couldn't parse into any of the possible types.

e.g. 
```rust
enum ChannelContents {
    OldChannelContent { ... }
    ChannelContent { /* the new one */ }
}    
```

And now that I write this, maybe we should using something like `ChannelContentV1` and `ChannelContentV2`, just in case we need to add another in future?

Reply:
let me take a look at this and see if i can get it to work